include/can.h

```diff
@@ -96,6 +109,16 @@ enum can_mode {
 	CAN_SILENT_LOOPBACK_MODE
 };
 
+/*
+ * Controller Area Network Identifier structure
+ *
+ * bit 0-28 : CAN identifier (11/29 bit)
+ * bit 29   : error message frame flag (0 = data frame, 1 = error message)
+ * bit 30   : remote transmission request flag (1 = rtr frame)
+ * bit 31   : frame format flag (0 = standard 11 bit, 1 = extended 29 bit)
+ */
+typedef u32_t canid_t;
+
```

Comment:
IMO its confusing to have two types of identifier now. canid_t and the can_msg itself. Maybe its better to offload the socketcan things to a socketcan.h where you include can.h?